Caching bundle install to improve build time

yamilurbina opened the pull request:
Requested on issue #151.

Running `bundle install` now saves the gems on a folder called `bundler_cache` on the same `api` folder during build. Next builds will restore this folder, resulting in saving time installing the gems.

MaxWofford commented:
Thanks for the PR @yamilurbina! Just assigned @paked to check it out.

paked commented:
Thanks @yamilurbina! This is an awesome PR.

Before I merge this, I just want to make sure that it works -- do you think you could set up your fork with Circle-CI to demonstrate the speed increase?

yamilurbina commented:
Hi @paked! I've added CircleCI to my fork but it's failing when running `bundle exec rubocop`. However, you can see that Bundler reuses the gems instead of installing them again: https://circleci.com/gh/yamilurbina/monolith/3

![screen shot 2017-10-01 at 9 48 03 pm](https://user-images.githubusercontent.com/304195/31061748-629bb9de-a6f2-11e7-8557-8f46ef5382c1.png)

paked commented:
Awesome! It's good to see that caching is working. Thanks for verifying that for me.

The reason why rubocop is failing now is because it's also checking our dependencies -- which don't follow rubocop's standards.

eg.

```

bundler_cache/ruby/2.3.0/gems/rake-12.0.0/doc/jamis.rb:2:1: C: Use 2 (not 0) spaces for indentation.
module Page

bundler_cache/ruby/2.3.0/gems/rake-12.0.0/doc/jamis.rb:4:1: C: Use 2 (not 0) spaces for indentation.
FONTS = "\"Bitstream Vera Sans\", Verdana, Arial, Helvetica, sans-serif"
```

I think what needs to happen is that we need to exclude the `bundler_cache` from rubocop's linting process. [This guide might be helpful](http://www.rubydoc.info/gems/rubocop/0.16.0#Including_Excluding_files)

yamilurbina commented:
@paked the build is now fixed, I added the `bundler_cache` folder to the list of ignored paths on `.rubocop.yml` file. https://circleci.com/gh/yamilurbina/monolith/4

paked commented:
Awesome job!

![https://media.giphy.com/media/NPTRgDfwjIZVK/giphy.gif](https://media.giphy.com/media/NPTRgDfwjIZVK/giphy.gif)

Thank you so much @yamilurbina :heart:
